### PR TITLE
Cleanup GridExternalPotential class

### PR DESCRIPTION
**Pull request by PhilipFackler**
## Proposed changes

Renamed and made private all data members
Refactored functions from h to cpp

## What type(s) of changes does this code introduce?
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 22.04

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCHamiltonians/GridExternalPotential.cpp [hunk 1]
```diff
@@ -18,6 +18,16 @@
 
 namespace qmcplusplus
 {
+GridExternalPotential::GridExternalPotential(ParticleSet& P) : ps_(P)
+{
+  setEnergyDomain(POTENTIAL);
+  oneBodyQuantumDomain(P);
+}
+
+std::string GridExternalPotential::getClassName() const { return "GridExternalPotential"; }
```
Comment (ye-luo) [inline src/QMCHamiltonians/GridExternalPotential.cpp:27]: Please keep this defined in the header file.

Comment (PhilipFackler) [reply src/QMCHamiltonians/GridExternalPotential.cpp]: done

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**